src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -263,4 +293,131 @@ describe("WalletOverviewPage", () => {
+    spyOn(comp, "initialize");
+
+    comp.ionViewDidLoad();
+
+    expect(comp.initialize).toHaveBeenCalled();
+  });
+
+  it("should have the funds visiblity to shown after they were invisible", () => {
+    comp.walletFundsVisibility = "hidden";
+
+    comp.fundsSwitch();
+
+    expect(comp.walletFundsVisibility).toBe("shown");
+    expect(comp.walletFundsVisibilityTransferButton).toBe("hidden");
+  });
+
+  it("should open the transfer page correctly", () => {
+    spyOn(navController, "push");
+
+    comp.openTransferPage();
+
+    expect(navController.push).toHaveBeenCalledWith(TransferPage);
+  })
+
+  it("should call balance and history after refreshing the wallet", () => {
+    spyOn(comp, "getWalletBalance");
+    spyOn(comp, "getTransactionHistory");
+
+    comp.currentWallet = <IWallet>{};
+    comp.currentWallet.publicKey = "";
+
+    comp.refreshWalletBalance();
+
+    expect(comp.getWalletBalance).toHaveBeenCalled();
+    expect(comp.getTransactionHistory).toHaveBeenCalled();
+  })
+
+  it("should not open landing page and show a toast after deleting a wallet while some wallets still exist", (done) => {
+    // Get all 3 wallets
+    comp.getAllWallets().then(data => {
+      // Delete first one
+      comp.deleteSelectedWallet(comp.currentWallet);
+
+      spyOn(comp, "openLandingPage");
+      spyOn(comp, "showToastMessage");
+
+      // Delete last one
+      comp.deleteSelectedWallet(comp.currentWallet);
+
+      expect(comp.openLandingPage).not.toHaveBeenCalled();
+      expect(comp.showToastMessage).not.toHaveBeenCalled();
+      
+      done();
+    });
+  })
+
+  it("should return 7 fixed numbers if the picked currency is not in the if condition", () => {
+    comp.pickedCurrency = "I DON'T EXIST";
+
+    let result = comp.getFixedNumbers();
+
+    expect(result).toEqual(7);
+  })
+
+  it("should call createElement, appendChild, execCommand and removechild upon copying to clipboard for web", () => {
+    spyOn(document, "createElement");
+    spyOn(document.body, "appendChild");
+    spyOn(document, "execCommand");
+    spyOn(document.body, "removeChild");
+
+    comp.copyToClipboardWeb("data");
+
+    expect(document.createElement).toHaveBeenCalled();
+    expect(document.body.appendChild).toHaveBeenCalled();
+    expect(document.execCommand).toHaveBeenCalled();
+    expect(document.body.removeChild).toHaveBeenCalled();
+  })
+
+  it("should call createElement, appendChild, removeChild upon downloading a txt file for web", () => {
+    spyOn(document, "createElement");
+    spyOn(document.body, "appendChild");
+    spyOn(document.body, "removeChild");
+
+    comp.downloadTxtFileWeb("data", "filename");
+
+    expect(document.createElement).toHaveBeenCalled();
+    expect(document.body.appendChild).toHaveBeenCalled();
+    expect(document.body.removeChild).toHaveBeenCalled();
+  });
+
+  it("should call create and present upon showing a toast message", () => {
+    spyOn(toastController, "create").and.returnValue(toast);
```

Comment:
As noted at the start of this review the `toast` object should be defined inside this function. Not globally.